final_task/rss_reader/action_functions.py

```diff
@@ -0,0 +1,118 @@
+""" Module of creation functions and action functions.
+
+    Functions:
+    create_logger(com_line_args) -> logger
+    get_news(command_line_args, logger) -> news_collection
+    print_news_stdout(news_collection) -> None
+    print_news_json(news_collection) -> None
+    print_news(news_collection, com_line_args, logger) -> None """
+
+import feedparser
+import json
+import logging
+from models import NewsEntry
+from validation_functions import check_limit_arg
+
+
+def create_logger(com_line_args):
+    """Create logger function.
+
+       Creates a logger considering the --verbose argument. """
+    # Create a logger
+    logger = logging.getLogger("rss_reader_logger")
+    logger.setLevel(logging.DEBUG)
+
+    # Create handlers
+    c_handler = logging.StreamHandler()
+    f_handler = logging.FileHandler("file.log")
+
+    # Check --verbose argument
+    if com_line_args.verbose:
+        c_handler.setLevel(logging.DEBUG)
+    else:
+        c_handler.setLevel(logging.ERROR)
+
+    # Create formatters and add it to handlers
+    c_format = logging.Formatter('%(name)s - %(levelname)s - %(message)s')
+    f_format = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+    c_handler.setFormatter(c_format)
+    f_handler.setFormatter(f_format)
+
+    # Add handlers to the logger
+    logger.addHandler(f_handler)
+    logger.addHandler(c_handler)
+
+    return logger
+
+
+def get_news(command_line_args, logger):
+    """ Get news function.
+
+        Uses feedparser library to receive news. """
+    logger.info("Getting news.")
+    news_feed = feedparser.parse(command_line_args.source)
+    news_collection = {}
+    feed = {"title": news_feed.feed.get("title", ""),
+            "date": news_feed.feed.get("published", ""),
+            "language": news_feed.feed.get("language", "")}
+
+    news_collection["feed"] = feed
+    news_collection["entries"] = []
+
+    for entry in news_feed.entries:
+        news_entry = NewsEntry()
```

Comment:
There is a misunderstanding of what attributes do you assign. Your class NewEntry has class attribute title, date and so on. You are working not with them, after creating of the instance of class NewEntry. You create new attributes title, date, link, summary, which reffer to instance. You never change class attributes. In order to make things work properly, please take a look at `dataclasses`.

Reply:
fixed